### DISCUSSION TITLE
A overloaded functions for HelloImGui::ImageFromAsset and ImageButtonFromAsset which can help us load and play GIF photo

### DISCUSSION THREAD

**Discussion by XMNXofficial** [pthom/hello_imgui#60] [Ideas]
# A overloaded functions
## What is this

I have created overloaded functions for `HelloImGui::ImageFromAsset` and `ImageButtonFromAsset`, which will aid in loading and playing `GIF images`. 

These functions are experimental and may contain several vulnerabilities, and I am currently working on improving them.

## How to use

```cpp
    HelloImGui::ImageFromAsset("test.gif", true);
    HelloImGui::ImageButtonFromAsset("test.gif", true);
```

## Some bug i can not fix
I am a beginner in C++, and I am not proficient in using C++😂😂.
- [ ] Can't  load some GIF images due to their high resolution.(DeBug:`stbi_load_gif_from_memory` return null)
- [ ] Gif play speed error.(Affected by `Power Save mode`.)

##  Show
[test.webm](https://github.com/pthom/hello_imgui/assets/112557849/f831fbcb-396b-4746-b3e0-5c2e0b440cfb)

**Comment by pthom** [comment]
Hi, this looks nice, thanks!

May be you could open a PR (once you are satisfied with your modifications), so that I can have a look at the code;

Thanks!

**Reply by XMNXofficial** [reply]
Ok~🥳

**Comment by pthom** [comment]
>  Gif play speed error.(Affected by Power Save mode.)

ImGui::GetTime() returns the current time in seconds. You could use this to advance, taking into account the framerate of the animation
